coderd/httpmw/apikey.go

```diff
@@ -237,6 +249,22 @@ func ExtractAPIKey(db database.Store, oauth *OAuth2Configs, redirectToLogin bool
 					})
 					return
 				}
+				if link.UserID != uuid.Nil {
```

Comment:
true, true

Reply:
actually this can be nil if the API key is not associated with a user_link, in other words, if it's a 'password' API key then there is no reference to a row in the `user_links` table.